### PR TITLE
Tab design improvements

### LINKED ISSUE #7150: Tab design improvements
# Current behavior

To open a tab: Open a note or task on a record, then press the minimize button:

CleanShot 2024-09-19 at 15 24 19

This reduce the side panel in a tab that look like this

CleanShot 2024-09-19 at 15 19 00

# Desired Behavior

Improve the tab design:
- Background should be transparent/secondary
- The tab should have a blur (like the one we use on menus)
- The tab should have a top left border-radius
- The tab should have some border
- Check the different spacings

image

https://www.figma.com/design/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=40315-524430&node-type=frame&t=VEnadEyAJpb29rmb-11

### PR DESCRIPTION
**Pull request by ArafatOSDev**
## What does this PR do?
Improve minimize tab design. 

Fixes #7150 

Screenshot 2024-09-28 at 11 46 35 AM

### CODE REVIEW

File: packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx [hunk 1]
```diff
@@ -30,11 +30,17 @@ import { RightDrawerRouter } from './RightDrawerRouter';
 
 const StyledContainer = styled(motion.div)`
   background: ${({ theme }) => theme.background.primary};
-  border-left: 1px solid ${({ theme }) => theme.border.color.medium};
+  border-top-left-radius: 8px;
+  border-left: 1px solid #d6d6d6;
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx:34]: We dont use hard-coded values!

File: packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx [hunk 2]
```diff
@@ -30,11 +30,17 @@ import { RightDrawerRouter } from './RightDrawerRouter';
 
 const StyledContainer = styled(motion.div)`
   background: ${({ theme }) => theme.background.primary};
-  border-left: 1px solid ${({ theme }) => theme.border.color.medium};
+  border-top-left-radius: 8px;
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx:33]: use theme.spacing(2)

File: packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx [hunk 3]
```diff
@@ -30,11 +30,17 @@ import { RightDrawerRouter } from './RightDrawerRouter';
...
+  border-left: 1px solid #d6d6d6;
+  border-top: 1px solid #d6d6d6;
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx:35]: Same here, no hard-coded values

File: packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx [hunk 4]
```diff
@@ -30,11 +30,17 @@ import { RightDrawerRouter } from './RightDrawerRouter';
...
   box-shadow: ${({ theme }) => theme.boxShadow.strong};
   height: 100dvh;
   overflow-x: hidden;
   position: fixed;
+  backdrop-filter: blur(40px);
+  box-shadow: 0px 2px 4px 0px #0000000A,
+              0px 0px 4px 0px #00000014;
```
Comment (ehconitin) [inline packages/twenty-front/src/modules/ui/layout/right-drawer/components/RightDrawer.tsx:42]: use theme.boxShadow.light

**Review by ehconitin — COMMENTED** [review]
Left some comments ;)

**Comment by ehconitin** [comment]
@FelixMalfait LGTM!
Could you check if animations are smooth, I couldnt judge with my laggy machine :)

**Comment by ehconitin** [comment]
@Bonapara 
should minimized tab also appear for mobile devices? 
As of now we dont show minimize button for mobile devices.

**Comment by ArafatOSDev** [comment]
Hi @FelixMalfait, I can keep the right tab bar design & change the minimize tab design.

**Comment by FelixMalfait** [comment]
@ArafatOSDev thanks! Would be great if you can make the required adjustments

**Comment by ehconitin** [comment]
@ArafatOSDev are you still working on this PR?

**Comment by ArafatOSDev** [comment]
@ehconitin Yes, I am working on this PR.

**Comment by ArafatOSDev** [comment]
@ehconitin Can you check this PR now?

**Comment by FelixMalfait** [comment]
Hey @ArafatOSDev I pushed a fix. We should avoid introducing new variables + avoid using useEffect unless it's strictly necessary. Also the border color was stronger in minimized state. I revert the commits to start fresh but took inspiration from your code. Thanks for your contribution!

**Review by FelixMalfait — APPROVED** [review]

**Comment by ehconitin** [comment]
Thanks @FelixMalfait

**Comment by ArafatOSDev** [comment]
Thanks @FelixMalfait I will follow your suggestion for the next contribution.

**Pull request merged**